### PR TITLE
fix InteractivePostStats feature when thread has guest users

### PR DESCRIPTION
**Pull request by yellow**
The InteractivePostStats feature fails to hyperlink stats after the `$(".post").each(function() {..}` loop encounters a post which has a guest/purged author.

Example:
https://hackforums.net/showthread.php?tid=353472
The feature breaks after the post by `Jonathannnnn`

added a basic try-catch-return block, could be done better
i have not tested it, but it should work ( i don't know JS)

reference:  https://stackoverflow.com/a/17162375

Fixes: #238

### CODE REVIEW

**Comment by yellow** [comment]
much better
thank you

**Review by xadamxk — APPROVED** [review]

File: src/features/threads/InteractivePostStats.js [hunk 1]
```diff
@@ -15,7 +15,8 @@ class InteractivePostStats extends Feature {
     // Loop post author
     $(".post").each(function() {
       const postbitTable = $(this).find(".post_author");
-      const userId = $(postbitTable).find(".author_information .largetext > a").attr("href").split("&uid=")[1];
+      const userId = $(postbitTable).find(".author_information .largetext > a").attr("href")?.split("&uid=")[1];
+      if (!userId) return;
```
Comment (xadamxk) [inline src/features/threads/InteractivePostStats.js:19]: Have you tested that returning null doesn't break the loop?
The jquery docs say returning false breaks the loop, so I wanted to be sure any users after the guest still function as expected. (Docs: https://api.jquery.com/jQuery.each/)

Comment (yellow) [reply src/features/threads/InteractivePostStats.js:19]: I have not tested it. I looked it up on StackOverflow https://stackoverflow.com/a/17162375
From what I gather:
`return false` == `break`
 `return` == `continue`

Comment (xadamxk) [reply src/features/threads/InteractivePostStats.js:19]: I'll test it out tomorrow after work, thanks!

**Comment by xadamxk** [comment]
@yellow PM me on site and I'll add the appropriate badges to your profile, thanks!

**Pull request merged**